backend/views/syson-tree-explorer-view/src/test/java/org/eclipse/syson/tree/explorer/view/services/SysONDefaultExplorerServicesTest.java

```diff
@@ -0,0 +1,108 @@
+import org.eclipse.syson.tree.explorer.view.services.api.ISysONExplorerFilterService;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+public class SysONDefaultExplorerServicesTest {
+
+    private static EditingContext editingContext;
+
+    private SysONDefaultExplorerServices sysONDefaultExplorerServices;
+
+    @BeforeAll
+    static void createEditingContext() {
+        ComposedAdapterFactory composedAdapterFactory = new ComposedAdapterFactory();
+        composedAdapterFactory.addAdapterFactory(new EcoreAdapterFactory());
+        composedAdapterFactory.addAdapterFactory(new ReflectiveItemProviderAdapterFactory());
+
+        EPackage.Registry ePackageRegistry = new EPackageRegistryImpl();
+        ePackageRegistry.put(EcorePackage.eINSTANCE.getNsURI(), EcorePackage.eINSTANCE);
+
+        AdapterFactoryEditingDomain editingDomain = new AdapterFactoryEditingDomain(composedAdapterFactory, new BasicCommandStack());
+        ResourceSet resourceSet = editingDomain.getResourceSet();
+        resourceSet.setPackageRegistry(ePackageRegistry);
+        resourceSet.eAdapters().add(new ECrossReferenceAdapter());
+        editingContext = new EditingContext(UUID.randomUUID().toString(), editingDomain, Map.of(), List.of());
+    }
+
+    @Test
+    @DisplayName("hasChildren method can handle non sysml content")
+    public void hasChildrenCanHandleNonSysmlContent() {
+        this.createMockServicesForHasChildren();
+
+        EPackage ePackage = EcoreFactory.eINSTANCE.createEPackage();
+        ePackage.setName("somePackage");
+        EClass c1 = EcoreFactory.eINSTANCE.createEClass();
+        ePackage.getEClassifiers().add(c1);
+        EClass c2 = EcoreFactory.eINSTANCE.createEClass();
+        ePackage.getEClassifiers().add(c2);
+        EAttribute c1_a1 = EcoreFactory.eINSTANCE.createEAttribute();
+        c1.getEStructuralFeatures().add(c1_a1);
+        EAttribute c1_a2 = EcoreFactory.eINSTANCE.createEAttribute();
+        c1.getEStructuralFeatures().add(c1_a2);
+
+        assertThat(sysONDefaultExplorerServices.hasChildren(ePackage, editingContext, List.of(), List.of())).isTrue();
+        assertThat(sysONDefaultExplorerServices.hasChildren(c1, editingContext, List.of(), List.of())).isTrue();
+        assertThat(sysONDefaultExplorerServices.hasChildren(c1_a1, editingContext, List.of(), List.of())).isFalse();
+    }
+
+    /**
+     * Create instances of services required to test hasChildren
+     * Use mock instances for non required services
+     */
+    private void createMockServicesForHasChildren() {
+
+        IIdentityService identityService = mock(IIdentityService.class);
```

Comment:
Thank you for this test but we don't use mockito in SysON.
Please use the NoOp versions of services.
Instead, you should be able to use:
-`IIdentityService identityService = new IIdentityService.NoOp();`
- `IContentService contentService = new IContentService.NoOp();`
- `IObjectService objectService = new IObjectService.NoOp();`
- `IURLParser urlParser = new IURLParser.NoOp();`

For `IRepresentationMetadataSearchService ` there is unfortunately no `NoOp` version of it.
So just create a `NoOp` version in this method or in this class and it should be ok.